### PR TITLE
[IMP] open_academy: Created session and partner models and related them with previously created course model; generated views to manage these relations T#53764

### PR DESCRIPTION
**Pull request by payen000**
A Session model was defined. Sessions relate to courses by their ids and have durations, instructors, attendees and starting dates.
image

A responsible user field was defined for the course model and put in the form view.
image

Partners had their views inherited from the res.partner view using view inheritance. 
image

Partners can manage the course sessions they will be attending to by clicking on the Open Academy sessions tab and adding a line from which the following window pops out:
image

### CODE REVIEW

**Comment by payen000** [comment]
@Jerudiel @rolandojduartem Could you review my code, please? :)

File: open_academy/models/session.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+
+class Session(models.Model):
+    _name = 'session'
+    _description = 'Session'
+
+    name = fields.Char(required=True)
+    start_date = fields.Date(string='Start date', required=True)
+    duration = fields.Float(string='Duration', required=True)
+    number_of_seats = fields.Integer(string='Seats', required=True)
```
Comment (Jerudiel) [inline open_academy/models/session.py:11]: I think is not required string parameter and required parameter is not necessary in all parameters, for this exercise it is not necessary for anyone

File: open_academy/views/session.xml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+            </tree>
+        </field>
+    </record>
+
+    <record id='session_action_display_view' model='ir.actions.act_window'>
+        <field name='name'>Session</field>
+        <field name='res_model'>session</field>
+        <field name='view_mode'>tree,form</field>
+    </record>
+</odoo>
```
Comment (Jerudiel) [inline open_academy/views/session.xml]: add new line, please (:

**Comment by rolandojduartem** [comment]
Also, I noticed a merge commit, probably you are going to need squash it.

File: open_academy/models/course.py [hunk 3]
```diff
@@ -8,3 +8,4 @@ class Course(models.Model):
     title = fields.Char(required=True)
     description = fields.Text()
     responsible_user_id = fields.Many2one('res.users')
+    session_ids = fields.One2many(comodel_name='session', inverse_name='id')
```
Comment (Jerudiel) [inline open_academy/models/course.py:11]: Inverse_name is the field name of model session, in this case is 'course_id'

File: open_academy/models/session.py [hunk 4]
```diff
@@ -11,3 +11,4 @@ class Session(models.Model):
     number_of_seats = fields.Integer(string='Seats', required=True)
     instructor_id = fields.Many2one('res.partner')
     course_id = fields.Many2one('course', required=True)
+    attendees_ids = fields.Many2many('res.partner')
```
Comment (Jerudiel) [inline open_academy/models/session.py:14]: Please replace 'attendees_ids' for 'attendee_ids'

File: open_academy/views/session.xml [hunk 5]
```diff
@@ -10,6 +10,7 @@
                 <field name='start_date'/>
                 <field name='duration' widget='float_time'/>
                 <field name='number_of_seats'/>
+                <field name='attendees_ids'/>
```
Comment (Jerudiel) [inline open_academy/views/session.xml:13]: Here again replace 'attendees_ids' for 'attendee_ids' (:

File: open_academy/views/inherited_partner.xml [hunk 6]
```diff
@@ -0,0 +1,31 @@
+<odoo>
+    <record id='partner_view_tree_inherited' model='ir.ui.view'>
+        <field name='name'>res.partner.view.tree.inherited</field>
+        <field name='model'>res.partner</field>
+        <field name='inherit_id' ref="base.view_partner_tree"/>
+        <field name='arch' type='xml'>
+            <field name='display_name' position='after'>
+                <field name='session_ids'/>
+            </field>
```
Comment (Jerudiel) [inline open_academy/views/inherited_partner.xml:9]: Missing add is_instructor field

File: open_academy/views/inherited_partner.xml [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+    <record id='partner_view_form_inherited' model='ir.ui.view'>
+        <field name='name'>res.partner.view.form.inherited</field>
+        <field name='model'>res.partner</field>
+        <field name='inherit_id' ref="base.view_partner_form"/>
+        <field name='arch' type='xml'>
+            <page name='internal_notes' position='after'>
+                <page string='Open Academy sessions'>
+                    <field name='session_ids'/>
+                </page>
```
Comment (Jerudiel) [inline open_academy/views/inherited_partner.xml:21]: Here again, missing add is_instructor field (:

File: open_academy/models/inherited_partner.py [hunk 8]
```diff
@@ -0,0 +1,8 @@
+from odoo import models, fields
+
+
+class InheritedPartner(models.Model):
+	_inherit = 'res.partner'
+
+	is_instructor = fields.Boolean('Is an instructor')
+	session_ids = fields.Many2many('session', 'id')
```
Comment (Jerudiel) [inline open_academy/models/inherited_partner.py:8]: I think it's not necessary another parameter in Many2many, 'session' is enough

**Pull request closed without merging**